pysat/tests/test_sw.py

```diff
@@ -0,0 +1,437 @@
+
+        del kp_inst, combo_in
+
+    def test_combine_kp_no_standard(self):
+        """Test combine_kp when standard data is not provided"""
+
+        if not self.download:
+            raise skip.SkipTest("test needs downloaded data")
+
+        combo_in = {kk: self.combine for kk in self.combine.keys()
+                    if kk != 'standard_inst'}
+        kp_inst = sw_methods.combine_kp(combo_in)
+
+        assert kp_inst.index[0] >= self.combine['recent_inst'].index[0]
+        assert kp_inst.index[-1] < self.combine['stop']
+        assert len(kp_inst.data.columns) == 1
+        assert kp_inst.data.columns[0] == 'Kp'
+        assert(kp_inst.meta['Kp'][kp_inst.meta.fill_label] ==
+               self.combine['fill_val'])
+        assert len(kp_inst['Kp'][kp_inst['Kp']]
+                   == self.combine['fill_val']) > 0
+
+        del kp_inst, combo_in
+
+
+class TestSWF107():
+    def setup(self):
+        """Runs before every method to create a clean testing setup"""
+        # Load a test instrument
+        self.testInst = pysat.Instrument()
+        self.testInst.data = pds.DataFrame({'f107': np.linspace(70, 200, 160)},
+                                             index=[pysat.datetime(2009, 1, 1)
+                                                    + pds.DateOffset(days=i)
+                                                    for i in range(160)])
+
+        # Set combination testing input
+        self.today = dt.datetime.today().date()
+        self.combineInst = {tag: pysat.Instrument("sw", "f107", tag)
+                            for tag in sw_f107.tags.keys()}
+        self.combineTimes = {"start": self.today - dt.timedelta(days=30),
+                             "stop": self.today + dt.timedelta(days=3)}
+
+        # Download combination testing input
+        self.download = True
+        # Load the instrument objects
+        for kk in self.combineInst.keys():
+            try:
+                if kk == '':
+                    self.combineInst[kk].download(self.combineTimes['start'],
+                                                  freq='MS')
+                else:
+                    self.combineInst[kk].download(**self.combineTimes)
+            except:
+                pass
+
+            if len(self.combineInst[kk].files.files) == 0:
+                self.download = False
+
+    def teardown(self):
+        """Runs after every method to clean up previous testing."""
+        del self.combineInst, self.download, self.today, self.combineTimes
+        del self.testInst
+
+    def test_combine_f107_none(self):
+        """ Test combine_f107 failure when no input is provided"""
+        
+        assert_raises(TypeError, sw_methods.combine_f107)
+
+    def test_combine_f107_no_time(self):
+        """Test combine_f107 failure when no times are provided"""
+
+        assert_raises(ValueError, sw_methods.combine_f107,
+                      [self.combineInst[''], self.combineInst['forecast']])
```

Comment:
self.combineInst[''], self.combineInst['forecast'])

each argument to combine_f107 needs to be on its own in the assert raises call

Reply:
Fixed.